src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/MulticlassLogisticRegression.cs

```diff
@@ -992,35 +950,280 @@ DataViewRow ICanGetSummaryAsIRow.GetSummaryIRowOrNull(RoleMappedSchema schema)
+
+        private static VersionInfo VersionInfo =>
+            new VersionInfo(
+                modelSignature: "MCLINEAR",
+                verWrittenCur: 0x00010001,
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(LinearMulticlassModelParameters).Assembly.FullName);
+
+        /// <summary>
+        /// Function used to pass <see cref="VersionInfo"/> into parent class. It may be used when saving the model.
+        /// </summary>
+        private protected override VersionInfo GetVersionInfo() => VersionInfo;
+
+        internal LinearMulticlassModelParameters(IHostEnvironment env, in VBuffer<float> weights, int numClasses, int numFeatures, string[] labelNames, LinearModelStatistics stats = null)
+            : base(env, RegistrationName, weights, numClasses, numFeatures, labelNames, stats)
+        {
+        }
+
+        internal LinearMulticlassModelParameters(IHostEnvironment env, VBuffer<float>[] weights, float[] bias, int numClasses, int numFeatures, string[] labelNames, LinearModelStatistics stats = null)
+            : base(env, RegistrationName, weights, bias, numClasses, numFeatures, labelNames, stats)
+        {
+        }
+
+        private LinearMulticlassModelParameters(IHostEnvironment env, ModelLoadContext ctx)
+            : base(env, RegistrationName, ctx)
+        {
+        }
+
+        /// <summary>
+        /// This function does not do any calibration. It's common in multi-class support vector machines where probabilitic outputs are not provided.
+        /// </summary>
+        /// <param name="dst">Score vector should be calibrated.</param>
+        private protected override void Calibrate(Span<float> dst)
+        {
+            return;
+        }
+
+        private static LinearMulticlassModelParameters Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
-            var host = env.Register("TrainLRMultiClass");
-            host.CheckValue(input, nameof(input));
-            EntryPointUtils.CheckInputArgs(host, input);
+            env.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(VersionInfo);
+            return new LinearMulticlassModelParameters(env, ctx);
+        }
 
-            return TrainerEntryPointsUtils.Train<LogisticRegressionMulticlassClassificationTrainer.Options, CommonOutputs.MulticlassClassificationOutput>(host, input,
-                () => new LogisticRegressionMulticlassClassificationTrainer(host, input),
-                () => TrainerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.LabelColumnName),
-                () => TrainerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.ExampleWeightColumnName));
+        /// <summary>
+        /// Output the text model to a given writer
```

Comment:
> Output the text model to a given writer [](start = 11, length = 40)

nit: dot in the end of sentence for all this comments in file.

Reply:
Thanks. I just checked both of this file and SdcaMulticlass.cs.

---
In reply to: [266993940](https://github.com/dotnet/machinelearning/pull/2976#discussion_r266993940) [](ancestors = 266993940)